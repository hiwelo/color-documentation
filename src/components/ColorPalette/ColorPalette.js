import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import defaultOptions from '../../config/defaultOptions';

const ColorPalette = props => <></>;

ColorPalette.propTypes = {
  /**
   * Specify a color palette to use with this component.
   * 
   * A color palette is an object composed of each color value with the name as key
   */
  palette: PropTypes.object,

  /**
   * Specify the palette name
   */
  name: PropTypes.string,

  /**
   * Specify this component options
   */
  options: PropTypes.object,
};

ColorPalette.defaultProps = {
  palette: {},
  name: undefined,
  options: defaultOptions,
}