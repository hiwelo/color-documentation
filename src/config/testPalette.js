const testPalette = {
  generic: {
    black: "#000",
    white: "#fff"
  }
};

export default testPalette;
