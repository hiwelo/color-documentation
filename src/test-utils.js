import { render } from '@testing-library/react';

/**
 * This file is compiling all useful methods
 * and consts used by unit tests.
 */

const queryByTestId = (testId) => {
  return document.querySelector(`[data-testid="${testId}"]`);
};

const queryParentByTestId = (testId) => {
  return queryByTestId(testId).parentElement;
};

const getTag = (element) => {
  return element.tagName.toLowerCase();
}

const doSnapshotTest = (DOMNode) => {
  const { container } = render(DOMNode);

  expect(container.firstChild).toMatchSnapshot();
};

export { queryByTestId, queryParentByTestId, getTag, doSnapshotTest };
