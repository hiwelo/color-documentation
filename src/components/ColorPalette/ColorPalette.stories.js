import React from 'react';
import { storiesOf } from '@storybook/react';
import ColorPalette from './ColorPalette';
import testPalette from '../../config/testPalette';

storiesOf('ColorPalette', module)
  .add('overview', () => (
    <ColorPalette name={'test'} colors={testPalette.generic} />
  ))
;
