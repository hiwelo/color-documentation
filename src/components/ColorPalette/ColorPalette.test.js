import React from 'react';
import { render } from '@testing-library/react';
import ColorPalette from './ColorPalette';
import { doSnapshotTest } from '../../test-utils.js';
import testPalette from '../../config/testPalette.js';

const defaultProps = {
  palette: testPalette[0],
  name: Object.keys(testPalette)[0],
}

describe('ColorPalette', () => {
  it('should render without crashing', () => {
    render(<ColorPalette {...defaultProps} />);
  });

  it('should match snapshot', () => {
    doSnapshotTest(<ColorPalette {...defaultProps} />);
  });
});
