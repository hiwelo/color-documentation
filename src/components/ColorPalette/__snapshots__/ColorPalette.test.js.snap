// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPalette should match snapshot 1`] = `
<article
  class="sc-bdVaJa bDWFJH"
>
  <h2
    class="sc-bwzfXH uTxCW"
  >
     generic palette
  </h2>
</article>
`;
